File: data structures and algorithms/python/06_079.py
def median_of_three(arr, low, high):
    mid = (low + high) // 2
    
    # Find median of first, middle, and last elements
    if arr[low] > arr[mid]:
        arr[low], arr[mid] = arr[mid], arr[low]
    if arr[mid] > arr[high]:
        arr[mid], arr[high] = arr[high], arr[mid]
    if arr[low] > arr[mid]:
        arr[low], arr[mid] = arr[mid], arr[low]
    
    # Return the median as pivot (usually at position high-1)
    arr[mid], arr[high-1] = arr[high-1], arr[mid]
    return arr[high-1]

File: data structures and algorithms/cs/06_079.cs
private static int MedianOfThree(int[] arr, int low, int high) {
   int mid = (low + high) / 2;
   
   // Find median of first, middle, and last elements
   if (arr[low] > arr[mid]) {
       Swap(arr, low, mid);
   }
   if (arr[mid] > arr[high]) {
       Swap(arr, mid, high);
   }
   if (arr[low] > arr[mid]) {
       Swap(arr, low, mid);
   }
   
   // Return the median as pivot (usually place it at position high-1)
   Swap(arr, mid, high-1);
   return arr[high-1];
}

private static void Swap(int[] arr, int i, int j) {
   int temp = arr[i];
   arr[i] = arr[j];
   arr[j] = temp;
}